rs/orchestrator/src/utils.rs

```diff
@@ -35,3 +39,140 @@ fn endpoint_to_url(protocol: &str, http: &ConnectionEndpoint, log: &ReplicaLogge
+            info!(log, "Detected node joining a subnet {subnet_id}");
+        }
+        (NodeAssignationState::Unassigned, NodeAssignationState::Assigned { subnet_id }) => {
+            // Maybe log instant transition...
+            info!(
+                log,
+                "Node instantly changed from Unassigned to Assigned to a subnet {subnet_id}"
+            );
+        }
+        (NodeAssignationState::Leaving { subnet_id }, NodeAssignationState::Unassigned) => {
+            // Maybe log node left subnet...
+            info!(log, "Node gracefully left a subnet {subnet_id}");
+            UtilityCommand::notify_host(&format!("The node {node_id} has gracefully left subnet {subnet_id}. The node can be turned off now."), 1);
+        }
+        (
+            NodeAssignationState::Joining {
+                subnet_id: joining_subnet,
+            },
+            NodeAssignationState::Assigned {
+                subnet_id: joined_subnet,
+            },
+        ) => {
+            if joining_subnet == joined_subnet {
+                // Maybe log node joined subnet...
+                info!(log, "Node joined subnet {joined_subnet}");
+            }
+        }
+        (NodeAssignationState::Assigned { subnet_id }, NodeAssignationState::Unassigned) => {
+            // Maybe log instant transition...
+            info!(
+                log,
+                "Node instantly changed from Assigned {subnet_id} to Unassigned"
+            );
+        }
+        (
+            NodeAssignationState::Assigned {
+                subnet_id: assinged_subnet,
+            },
+            NodeAssignationState::Leaving {
+                subnet_id: leaving_subnet,
+            },
+        ) if assinged_subnet == leaving_subnet => {
+            // Maybe log leaving subnet...
+            info!(log, "Node started leaving subnet {assinged_subnet}");
+            UtilityCommand::notify_host(&format!("The node {node_id} has been unassigned from the subnet {assinged_subnet} in the registry. Please do not turn off the machine while it completes its graceful removal from the subnet. This process can take up to 15 minutes. A new message will be displayed here when the node has been successfully removed."), 1);
+        }
+        // TODO: ATM this is not possible but it may be in the future
+        (
+            NodeAssignationState::Assigned {
+                subnet_id: assigned_subnet,
+            },
+            NodeAssignationState::Moving { from, to: _ },
+        ) if assigned_subnet == from => {
+            // Maybe log moving subnet started...
+        }
+        // TODO: ATM this is not possible but it may be in the future
+        (
+            NodeAssignationState::Moving { from: _, to },
+            NodeAssignationState::Assigned {
+                subnet_id: assigned_subnet,
+            },
+        ) if to == assigned_subnet => {
+            // Maybe log moving subnet completed...
+        }
```

Comment:
This might be an interesting feature in the future actually. It would be cool to be able to move a node from subnet A -> B without the need to initially unassign it. I left it here because it is a sensible state change, but we can remove it ofc.